Change build path of binaries to build folder

ersonp opened the pull request:
Did you run `make format && make check`?

Fixes #	

 Changes:	
- build path of binaries moved from `root` to `./build`
- added new makefile rules

How to test this PR:
1. Run `make build`
2. Run `cd ./build`
3. Run `./skywire-cli config gen -ir`
4. Run `./skywire-visor`
5. Check if it works on Mac, Linux and Windows
6. Check other related makefile rules
7. Test Autoconfig
8. Run `make build1` for old functionality i.e. build binaries in root

ersonp committed:
Change the build directory to /build

ersonp committed:
Update new binary path in readme

ersonp committed:
Update new binary path in package installer script

ersonp committed:
Update gitignore to use add build folder

ersonp committed:
Update readme

ersonp committed:
Update dockerfile

ersonp committed:
Update skyenv

ersonp committed:
Add make rules

This commit adds the rules unfix-systray-vendor and bin-fix.

This content of the rule bin is moved to bin-fix and the rule unfix-systray-vendor reverts the changes made by fix-systray-vendor.

The rule bin now only uses fix-systray-vendor bin-fix unfix-systray-vendor as dependencies in that order.

ersonp committed:
Fix make rule install-system-linux

mrpalide commented on pkg/skyenv/skyenv.go:
```diff
@@ -78,7 +78,7 @@ const (
 	AppSrvAddr                = "localhost:5505" // AppSrvAddr ...
 	ServiceDiscUpdateInterval = time.Minute      // ServiceDiscUpdateInterval ...
-	AppBinPath                = "./apps"         // AppBinPath ...
+	AppBinPath                = "./build/apps"   // AppBinPath ...
```
I think we don't need change these, and we just want `make build` command output move from root to  build folder.

mrpalide commented on cmd/skywire-cli/README.md:
```diff
@@ -318,7 +318,7 @@ $ skywire-cli config gen -bpirxn --version 1.3.0
 		"apps": null,
 		"server_addr": "localhost:5505",
-		"bin_path": "./apps",
+		"bin_path": "./build/apps",
```
Same as skyenv.go

mrpalide reviewed (commented):
Just move build output to build folder. And same as before, if user want run should go to build folder and do same as before every step, like generate config and run and etc.

ersonp committed:
Revert AppBinPath

ersonp committed:
Revert Readme change

ersonp committed:
Update readme

ersonp committed:
New makefile rule

This commit adds two new rules to the makefile bin1 and bin-root-path. bin1 is used to build the binaries in root.

The bin path is also now stored in a vairable called BUILD_PATH.

ersonp committed:
Add apps to gitignore

ersonp committed:
Fix target specific BUILD_PATH var

ersonp committed:
Add appveyor config for deploy

0pcom committed:
change config default for AppBinPath to ./build/apps from ./apps

0pcom committed:
make format test

0pcom committed:
fix makefile comments / improve 'make help'
